packages/flutter/lib/src/widgets/committed_content.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// A class representing data for the `onContentCommitted` callback on text fields.
+///
+/// This will be used when content is inserted into a text field. The class holds
+/// information for the mime type, URI (location), and bytedata for the inserted
+/// content.
+@immutable
+class CommittedContent {
+  /// Creates an object to represent content that is committed from a text field.
+  ///
+  /// Any parameters can be null.
+  const CommittedContent({this.mimeType, this.uri, this.data});
+
+  /// Converts Map received from Flutter Engine into the Dart class.
+  CommittedContent.fromMap(Map<String, dynamic>? metadata):
+      mimeType = metadata != null && metadata.isNotEmpty ? metadata['mimeType'] as String? : null,
+      uri = metadata != null && metadata.isNotEmpty ? metadata['uri'] as String? : null,
+      data = metadata != null && metadata.isNotEmpty ? Uint8List.fromList(
+          List<int>.from(metadata['data'] as Iterable<dynamic>)) : null;
+
+  /// Mime type of inserted content.
+  final String? mimeType;
```

Comment:
nit: full sentence e.g.: the mime type of the inserted content. Same for other fields.